### PR TITLE
Separate modules for each assembly

### PR DESCRIPTION
**Pull request by joshski**
Break apart `/features/support/world.js` into separate modules for each assembly. Each assembly is loaded in a generic way and conforms to a uniform interface including async start/stop.

Pros:
* Less code in one file
* Likely to scale better
* More obvious dependencies
* Assemblies have clear start and stop semantics

Cons:
* More code in total
* More indirection
* A little bit of magic

Since there are now explicit semantics for start and stop, it's nearly possible to make the getters for context/action/outcomeTodoList synchronous instead of async. The Dom one is kind of "late bound" though...

Also renamed `webdriver-http-database` to `webdriver-database` although perhaps this is less controversial and should be on master?

### CODE REVIEW

**Comment by joshski** [comment]
Also @aslakhellesoy I have no idea about this circle / sqlite3 thing. Can't see what this branch changes that could possibly break that. Works on my machine! Any ideas?

**Comment by joshski** [comment]
I guess one thing it does is loads sqlite _later_ because sequelize is required later...

**Comment by joshski** [comment]
Hey @aslakhellesoy what do you think? I won't be offended! 😃

File: features/support/assemblies/dom-http-memory.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  async actionTodoList() {
+    return await this.makeDomTodoList()
+  }
+
+  async outcomeTodoList() {
+    return await this.makeDomTodoList()
+  }
+
+  async makeDomTodoList() {
```
Comment (aslakhellesoy) [inline features/support/assemblies/dom-http-memory.js]: I like to use `_` as a prefix for private methods, to make it apparent they're not part of the public API/contract.

File: features/run/all [hunk 2]
```diff
@@ -5,5 +5,5 @@ features/run/http-memory && \
 features/run/dom-http-memory && \
 features/run/database && \
 features/run/webdriver-memory && \
-features/run/webdriver-http-database && \
```
Comment (aslakhellesoy) [inline features/run/all]: If you look at the lego drawing, you'll see that the assemblies are currently named after the `_|⎻|_|⎻|_` legos. I quite like that as a naming convention - it summarises what legos are in the stack.

File: features/support/assemblies/dom-http-memory.js (hunk 1 above)
Comment (joshski) [reply features/support/assemblies/dom-http-memory.js]: 👍

File: features/run/all (hunk 2 above)
Comment (joshski) [reply features/run/all]: Yeah I realised after doing this, we had already renamed to the opposite. What is confusing though, is that the `HttpTodoList` is not mentioned _in the assembly_ but it is, of course, used in the stack under test.

Comment (joshski) [reply features/run/all]: Looking at it again, I'm confused. Why isn't `webdriver-memory` called `webdriver-dom-memory` then? and why isn't `webdriver-http-database` called `webdriver-dom-http-database` then? That would seem to be the naming convention you are suggesting?!

screen shot 2017-11-07 at 17 09 02

Comment (joshski) [reply features/run/all]: Sticking with your names for now, so I can merge this. I'll open an issue to discuss further

**Pull request merged**